src/contest_link_80FC4F4.c

```diff
@@ -72,22 +72,22 @@ static void sub_80FC5DC(u8 taskId)
 
     gContestPlayerMonIndex = GetMultiplayerId();
     gNumLinkContestPlayers = GetLinkPlayerCount();
-    gIsLinkContest = 1;
+    gLinkContestFlags = LINK_CONTEST_FLAG_IS_LINK;
     if (gWirelessCommType == 1)
-        gIsLinkContest = 3;
+        gLinkContestFlags = LINK_CONTEST_FLAG_IS_LINK | LINK_CONTEST_FLAG_IS_WIRELESS;
 
     for (i = 0; i < gNumLinkContestPlayers && (u32)(gLinkPlayers[i].version & 0xFF) - 1 > VERSION_RUBY - 1; i++)
```

Comment:
You should be able to remove both those subtractions and cast and still have things work.

Reply:
No, it fails to match without the cast and subtractions. Both are necessary.